### PR TITLE
Document limitations of InPredicate-dependent queries

### PR DESCRIPTION
**Pull request by ArturGajowy**
This should be reverted after #6991 is resolved.

### CODE REVIEW

**Comment by ArturGajowy** [comment]
@electrum back to you

**Comment by electrum** [comment]
Is this saying that the query will fail if the values are null (to prevent a correctness issue)? Or that there is a bug and you get the wrong result if they are null? If this is warning about a bug, it should be a warning, not a note.

**Comment by ArturGajowy** [comment]
@electrum this is a warning about a query type that will fail during execution to prevent incorrect results

> This limitation is needed to ensure correctness and might be dropped in the future.

This is a follow-up for https://github.com/prestodb/presto/issues/6622#issuecomment-269804934

**Comment by ArturGajowy** [comment]
@electrum I made it explicit that the query will fail to make things more clear

**Comment by ArturGajowy** [comment]
@electrum kindly ping :)

**Comment by ArturGajowy** [comment]
@electrum kindly ping (icmp_seq=1) ;)

**Comment by ArturGajowy** [comment]
@electrum kindly ping, this is a minor doc change

**Comment by ArturGajowy** [comment]
@electrum kindly ping

File: presto-docs/src/main/sphinx/functions/comparison.rst [hunk 1]
```diff
@@ -153,3 +153,7 @@ Expression              Meaning
 ====================    ===========
 
 ``ANY`` and ``SOME`` have the same meaning and can be used interchangeably.
+
+.. note:: Currently the expression ``A`` in ``A = ANY (...)`` and ``A <> ALL (...)``
```
Comment (electrum) [inline presto-docs/src/main/sphinx/functions/comparison.rst]: Multi-line note blocks should use a separate block:
```
.. note::

    xxx
    xxx

Comment (electrum) [inline presto-docs/src/main/sphinx/functions/comparison.rst]: Comma after "Currently"

**Comment by electrum** [comment]
Merged, thanks!

**Pull request closed without merging**